core/silkworm/chain/config.cpp

```diff
@@ -71,12 +73,18 @@ nlohmann::json ChainConfig::to_json() const noexcept {
     member_to_json(ret, "daoForkBlock", dao_block);
     member_to_json(ret, "muirGlacierBlock", muir_glacier_block);
     member_to_json(ret, "arrowGlacierBlock", arrow_glacier_block);
+    member_to_json(ret, kTerminalBlockNumber, terminal_block_number);
 
     if (terminal_total_difficulty.has_value()) {
         // TODO (Andrew) geth probably treats terminalTotalDifficulty as a JSON number
         ret[kTerminalTotalDifficulty] = to_string(*terminal_total_difficulty);
     }
 
+    if (terminal_block_hash.has_value()) {
+        Bytes terminal_block_hash_bytes(kHashLength, '\0');
+        std::memcpy(&terminal_block_hash_bytes[0], terminal_block_hash->bytes, kHashLength);
```

Comment:
There's no need to create a temporary copy here. Just call `to_hex(*terminal_block_hash)`.